lib/client.js

```diff
@@ -23,6 +23,15 @@ var _defaultOptions = {
     certificate:  process.env.TASKCLUSTER_CERTIFICATE
   },
 
+  timeout: 30 * 1000, // 30 seconds
+
+  // Http agent options (https://nodejs.org/api/http.html#http_new_agent_options)
+  agent: {
+    keepAlive: true,
+    // XXX: This number is a random guess...
+    maxSockets: 256
```

Comment:
Also `maxSockets: 256` might be a bit high... Considering that node 0.10 had a limit of `5`.
Maybe `25` is enough... This is certainly an easy way to overload the server, because all the requests arrive at the same time.

Granted `5` seems a bit low...